packages/web-app-files/src/services/archiver.ts

```diff
@@ -69,8 +70,28 @@ export class ArchiverService {
     const url = options.publicToken
       ? downloadUrl
       : await clientService.owncloudSdk.signUrl(downloadUrl)
-    window.location = url
-    return url
+
+    try {
+      const { data } = await clientService.httpUnAuthenticated.get(url, {
+        headers: {
+          ...(!!options.publicLinkPassword && {
+            Authorization:
+              'Basic ' +
+              Buffer.from(['public', options.publicLinkPassword].join(':')).toString('base64')
+          })
+        },
+        responseType: 'arraybuffer'
+      })
+      const blob = new Blob([data], { type: 'application/tar' })
+      const objectUrl = URL.createObjectURL(blob)
+      const a = document.createElement('a')
+      a.download = 'download.tar'
+      a.href = objectUrl
+      a.click()
+      return objectUrl
```

Comment:
worth extracting this into a generic helper? Or do we even already have something generic?